ViAn/GUI/mainwindow.cpp

```diff
@@ -700,21 +678,58 @@ void MainWindow::on_actionChoose_Workspace_triggered() {
     /**
  * @brief MainWindow::on_actionDeleteProject_triggered
  * Deletes the saved files of the selected project.
- * Removes the project from the preoject tree.
+ * Removes the project from the project tree.
  */
 void MainWindow::on_actionDeleteProject_triggered() {
-    if(selectedProject != nullptr) {
-        QMessageBox::StandardButton resBtn = QMessageBox::question( this, "Delete",
-                                                                    tr("Are you sure you want to delete the selected project?\n"),
-                                                                    QMessageBox::No | QMessageBox::Yes,
-                                                                    QMessageBox::No);
-
-        if (resBtn == QMessageBox::Yes) {
-            this->fileHandler->delete_project(fileHandler->get_project(this->selectedProject->id));
-            remove_selected_project_from_tree();
+    QTreeWidgetItem *project;
+    MyQTreeWidgetItem *my_project;
+    if(ui->ProjectTree->selectedItems().size() == 1) {
+        project = ui->ProjectTree->selectedItems().first();
+        if (!project->parent()){
+            QMessageBox::StandardButton resBtn = QMessageBox::question( this, "Delete",
+                                                                        tr("Are you sure you want to delete the selected project?\n"),
+                                                                        QMessageBox::No | QMessageBox::Yes,
+                                                                        QMessageBox::No);
+
+            if (resBtn == QMessageBox::Yes) {
+                my_project = (MyQTreeWidgetItem*)project;
+                this->fileHandler->delete_project(fileHandler->get_project(my_project->id));
+                remove_selected_project_from_tree();
+            }
+        }else{
+            set_status_bar("No project selected to remove.");
         }
-    } else {
-        set_status_bar("No selected project to remove.");
+    }else{
+        set_status_bar("Multiple or no projects selected.");
```

Comment:
See if you can rewrite this code. It would be good to avoid nested if-statements. We should definately not have more than 2 nested if statements.